### PR TITLE
feat: mimo model support

### PR DESCRIPTION
**Pull request by Mozoltov821**
## Motivation

support XiaoMi-mimo model 

## Accuracy Tests
launcher server
```
 uv run python3 -u -m sgl_jax.launch_server   --model-path XiaomiMiMo/MiMo-7B-RL   --trust-remote-code   --device=tpu    --mem-fraction-static=0.8   --max-prefill-tokens=4096   --max-running-requests=128  --attention-backend=fa   --dtype=bfloat16  --port 30000 --host 0.0.0.0 --tp-size 4 --page-size 64
```
run eval test
```
uv run evalscope eval   --model XiaomiMiMo/MiMo-7B-RL   --api-url http://127.0.0.1:30000/v1/chat/completions   --datasets math_500   --eval-type service   --generation-config do_sample=true,temperature=0.6   --generation-config bos_token_id=151643,do_sample=false,eos_token_id=151643,max_new_tokens=2048,transformers_version=4.37.0 --eval-batchsize 128
```

launcher server with mtp
```
uv run python3 -u -m sgl_jax.launch_server   --model-path XiaomiMiMo/MiMo-7B-RL   --trust-remote-code   --device=tpu    --mem-fraction-static=0.5   --max-prefill-tokens=4096   --max-running-requests=16  --decode-log-interval=1   --enable-request-time-stats-logging   --attention-backend=fa   --dtype=bfloat16  --port 30000 --host 0.0.0.0 --disable-overlap-schedule --tp-size 4 --speculative-algorithm EAGLE --speculative-draft-model-path  XiaomiMiMo/MiMo-7B-RL --page-size 64 --speculative-eagle-topk 1 --speculative-num-steps 3 --speculative-num-draft-tokens 4 
```
image

## Checklist

- [x] Please use English, otherwise it will be closed.
- [x] The purpose of the PR, or link existing issues this PR will resolve.
- [x] The test plan, such as providing test command.
- [x] (Optional) The necessary documentation update.

### CODE REVIEW

File: python/sgl_jax/srt/models/mimo.py [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+                    config=config,
+                    layer_id=layer_id,
+                    dtype=dtype,
+                    rngs=rngs,
+                    mesh=mesh,
+                )
+                for layer_id in range(num_layers)
+            ]
+        self.mtp_layers = nnx.data(mtp_layers)
+        self._mtp_layers_initialized = True
```
Comment (pengchengneo) [inline python/sgl_jax/srt/models/mimo.py]: simply this funciton , move MiMoMTPLayer's init into MiMo Model init

File: .gitignore [hunk 2]
```diff
@@ -241,3 +241,5 @@ CLAUDE.md
 
 #gemini code
 .gemini-clipboard
+/docs/docker.md
+/Dockerfile
```
Comment (pengchengneo) [inline .gitignore]: what's purpose of this modification ?

**Comment by SiqiLi-Fighting** [comment]
please past your launch server command and precision benchmark result by evalscope here to make it is aligned with MiMo's paper result

File: .gitignore (hunk 2 above)
Comment (Mozoltov821) [reply .gitignore]: These two files are used in my local development and are not yet complete, so I ignored them. In the latest code, these two files are no longer ignored.

File: python/sgl_jax/srt/models/mimo.py (hunk 1 above)
Comment (Mozoltov821) [reply python/sgl_jax/srt/models/mimo.py]: It has been simplified.

File: python/sgl_jax/srt/kernels/ragged_paged_attention/ragged_paged_attention.py [hunk 3]
```diff
@@ -1361,6 +1361,9 @@ def ragged_paged_attention(
             max_num_tokens,
             pages_per_seq,
         )
+        if not causal:
+            bkv_p = 4
+            bq_sz = 4
```
Comment (JamesBrianD) [inline python/sgl_jax/srt/kernels/ragged_paged_attention/ragged_paged_attention.py]: too hack

Comment (pengchengneo) [reply python/sgl_jax/srt/kernels/ragged_paged_attention/ragged_paged_attention.py]: currently,  we still have no better choice here to avoid 'custom_mask mask vmem oom',  we need to optimize it future

Comment (JamesBrianD) [inline python/sgl_jax/srt/kernels/ragged_paged_attention/ragged_paged_attention.py]: Maybe hack code placed in `get_tuned_block_sizes.py`
